### PR TITLE
[Tooling] Remove zalpha flavor and alpha versioning

### PR DESCRIPTION
**Pull request by AliSoftware**
> **Note**: This PR is targeting the `release/20.5` branch, since I'll benefit from this change especially when I'll do upcoming betas.

## Why

Some more cleanup following the action of us removing the alpha builds when we migrated to Buildkite — [internal ref: paaHJt-3EU-p2]

## How

- Remove `alpha.version{Name,Code}` properties from `version.properties`
- Remove the `zalpha` flavor from `WordPress/build.gradle`
- Rearrange flavors versioning, so that all flavors use the same values of the `versionName` and `versionCode` properties from `version.properties` file (as opposed to some of them using `alpha.version{Name,Code}` before)

## To Test

 - Run `./gradlew printVersionName` and verify it prints the current `versionName` (`20.5-rc-1`)
 - Run `./gradlew printAllVersions` and verify it prints the current `versionName` and `versionCode` of each `{wordpress,jetpack}{Vanilla,Wasabi,Jalapeno}{Debug,Release}` flavor combination (should be `20.5-rc-1 (1259)` for all of them, now that we've unified them all to be the same on all flavors)
 - Test that the version bump still works with our tooling even without an alpha:
    - Cut a test branch named `release-test-17023` from this branch (the test branch **has** to be prefixed `release` for the prechecks of the next step to not reject you)
    - Run `PROJECT_ROOT_FOLDER=. bundle exec fastlane run android_bump_version_beta`
    - Verify that it pushed a new commit named `Bump version number` on your test branch, with `versionName` bumped to `20.5-rc-2` and `versionCode` bumped to `1260` in the `version.properties` file
    - 🧹  Delete your test `release-test-17023` branch, **both locally and from the remote** (since that test `git push`'d it 😒 )

## Additional thoughts to ponder for reviewers

Might also be worth confirming that this `versionCode` uniformization will not impact everyday development, for developers installing the apk on their test devices (and having to uninstall previous versions when `versionCode` is not increasing).

Personally I don't think that change will impact this — not only because developers mostly always work on branches cut from `trunk` and thus will still have increasing `versionCodes`, but also most of them test with the same flavor each time. But just thought it would be still be useful to raise the question to have your thoughts on it, in case I missed a use case.

### CODE REVIEW

**Review by oguzkocer — APPROVED** [review]
Since we removed the alpha flavor, the initial version bump will be to `versionCode=1260` which was already used in Play Store for the current alpha build. So, it'll need to be manually adjusted. I don't think it's important enough to block this PR, so I am just leaving that as an FYI to hopefully save you some time due to the upload failure.

> Might also be worth confirming that this versionCode uniformization will not impact everyday development, for developers installing the apk on their test devices (and having to uninstall previous versions when versionCode is not increasing).

One way to simplify this, and address the `1260` version code issue, could be to do a version bump without doing a beta release or hold off on merging these changes to `trunk` until we have a new beta or final release. After that, since we used the same `applicationId` for `alpha` & `vanilla` flavors, `vanilla` should override the existing `alpha` builds.

If we merge these changes to `trunk` now and developers use the `vanilla` build while having the latest `alpha` installed, they'll get a `INSTALL_FAILED_VERSION_DOWNGRADE` error. I don't think that's a problem, but if there is a way to avoid this confusion, I'd prefer to take it.

---

@AliSoftware Note that I disabled auto-merge, so you'll get a chance to take action about the `versionCode` issue if you would like to. However, please feel free to merge it manually if you want to land it as is.

**Comment by AliSoftware** [comment]
> Since we removed the alpha flavor, the initial version bump will be to versionCode=1260 which was already used in Play Store for the current alpha build. So, it'll need to be manually adjusted.

Oh that's a good point, I thought about mid/long term impact of that change but not the immediate impact at the moment we transition… 👍🙇🏻

But in practice, `1260` has never been published to PlayStore because we stopped generating (and thus submitting) alpha builds since `alpha-377` / versionCode `1252` (which was done alongside `20.3-rc-2` beta). Since then we only submitted betas, and the `alpha.versionCode`, despite being bumped in `version.properties` by our tooling because the entry was still there, was not used anymore to produce and upload alpha builds to PlayStore.

EB7971C6-1106-4F93-BD72-98F009240927

---

That being said, developers might still encounter a `versionCode` decrementing if they happened to have built à `zalpha` or `wasabi` build locally with their AndroidStudio, which used `alpha.versionCode = 1260` so far, but will use `1259` once this lands…

So even if it won't affect PlayStore, I agree that maybe it could still be worth just manually updating the value of `versionCode` to `1260` in `version.properties` (no need to _actually_ do a new beta build) to avoid that decrementing once the PR lands.

@oguzkocer  Do you think that simple manual bump would be enough to address all the concerns and points you highlighted above?

And again, good thoughts and thanks for thinking about that case! 🙌

File: version.properties
Comment (AliSoftware) [inline version.properties]: Following the good point made by @oguzkocer  in his review comment, I think just manually updating that line would be enough to avoid developers risking to encounter a `versionCode` decrease when this PR will land (since  `1260` was already previously used by `wasabi` flavor)

```suggestion
versionCode=1260
```

Comment (AliSoftware) [reply version.properties]: PS: I plan to merge `release/20.5` into `trunk` once this PR lands, so that the last 3 tooling-only PRs that happened in `release/20.5` recently would land in `trunk` before I even do the next beta (which would also help isolating those tooling PR landings from the future post-beta PR that will happen next later in the sprint when a new beta will be needed)

Comment (oguzkocer) [reply version.properties]: This works for me.

I don't like having a disconnection between the `versionName` and `versionCode`, so I was a bit hesitant to suggest this standalone bump. However, I admit that it's the most practical way to address the issue. Maybe you can leave a note in the commit message about why there is a separate bump for `versionCode` and that the artifact for it doesn't exist in Play Store, just as a breadcrumb in case we land in this commit while investigating the git history?

> no need to actually do a new beta build

Yeap, no need for a new beta, this bump is just to simplify the developer experience.

> and again, good thoughts and thanks for thinking about that case! 🙌

I am glad to hear that you found it useful 🙂

Comment (AliSoftware) [reply version.properties]: > I don't like having a disconnection between the versionName and versionCode, so I was a bit hesitant to suggest this standalone bump.

Yeah, I agree that  I'm not a fan either of having the `versionName` and `versionCode` in `version.properties` not match to the actual values we used for the latest upload of a build to the PlayStore.

> However, I admit that it's the most practical way to address the issue. 

Yeah I think this is acceptable to make addressing this easier indeed. We can live we that especially since it will re-sync next time we do a beta — which will be Wednesday next week… at the latest)

> Maybe you can leave a note in the commit message about why there is a separate bump for versionCode and that the artifact for it doesn't exist in Play Store, just as a breadcrumb in case we land in this commit while investigating the git history?

Totally 👍 . Did that in [020439b](https://github.com/wordpress-mobile/WordPress-Android/pull/17023/commits/020439bdd505c10200ffecbbd2c029d4dea1cdfe); hopefully I explained the whole thing well enough in that commit message.

**Pull request merged**
